packages/block-editor/src/components/list-view/block-select-button.js

```diff
@@ -189,6 +194,30 @@ function ListViewBlockSelectButton(
 					updateFocusAndSelection( updatedBlocks[ 0 ], false );
 				}
 			}
+		} else if ( isMatch( 'core/block-editor/insert-before', event ) ) {
```

Comment:
The logic for the list view needs to be slightly different to the editor canvas, as the list view does a couple of things differently:

* We determine the blocks to update based on the focused item in addition to the block selection (so if the focused list view item is outside of the selection, we ignore the selection and use the focused list view item as the "anchor" block to add blocks either before or after)
* After insertion, we switch focus based to the list view item so that focus remains within the list view

So, I'm not too sure how feasible it'd be to unify this further with how the editor canvas handles things. That said, it could be worth exploring, and to look into how to either neaten things or consolidate things further in a separate PR as this `onKeyDownHandler` callback is quite long, so if nothing else it might work to move the callback into a hook? In any case, I'll merge this PR in its current form for now, and we can look into code quality follow-ups separately 🙂